tests/cloner_test.go

```diff
@@ -508,6 +515,40 @@ var _ = Describe("all clone tests", func() {
 				err = utils.DeleteVerifierPod(f.K8sClient, f.Namespace.Name)
 				Expect(err).ToNot(HaveOccurred())
 			})
+
+			It("[rfe_id:1126][crit:High][vendor:cnv-qe@redhat.com][level:component] Should fail with Event when cloning into a smaller sized data volume", func() {
+				By("Creating a source from a real image") // todo fs....
+				sourceDv := utils.NewDataVolumeWithHTTPImport("source-dv", "200Mi", tinyCoreIsoURL())
+				filesystem := v1.PersistentVolumeFilesystem
+				sourceDv.Spec.PVC.VolumeMode = &filesystem
+				sourceDv, err := utils.CreateDataVolumeFromDefinition(f.CdiClient, f.Namespace.Name, sourceDv)
+				fmt.Fprintf(GinkgoWriter, "sourceDv %v", sourceDv)
+
+				Expect(err).ToNot(HaveOccurred())
+				f.ForceBindPvcIfDvIsWaitForFirstConsumer(sourceDv)
+
+				By("Waiting for import to be completed")
+				utils.WaitForDataVolumePhaseWithTimeout(f.CdiClient, f.Namespace.Name, cdiv1.Succeeded, sourceDv.Name, 3*90*time.Second)
+
+				By("Cloning from the source DataVolume to under sized target")
+				targetDv := utils.NewDataVolumeForImageCloningAndStorageSpec("target-dv", "100Mi",
```

Comment:
not sure if it is possible but wasn't the bug found because the requested size was the same size as the src but the fsoverhead wasnt taken into acount? putting 200Mi will work?

Reply:
So let me enumerate tests added or changed:
1. A new test for the BUG: cloner_test.go:472
`It("Should clone data from fs to fs while using calculated storage size", func() {`
2. A new test for cloning to smaller data volume with the use of Storage.spec (as it is now bypassing the validation hook) cloner_test.go:519:
			It("[rfe_id:1126][crit:High][vendor:cnv-qe@redhat.com][level:component] Should fail with Event when cloning into a smaller sized data volume", func() {
3. Very small cleanup for the existing test  cloner_test.go:886:
`		It("[rfe_id:1126][test_id:1896][crit:High][vendor:cnv-qe@redhat.com][level:component] Should not allow cloning into a smaller sized data volume", func() {`

Reply:
I understand I was wandering about (2) if using the same size 200Mi will work? should it work? or the 200Mi is doesnt include fsoversize so it shouldnt work and if so it should be tested

Reply:
Yeah I am fine with that, that should be an interesting test but outside of the scope of this PR I think.